### PR TITLE
Added tests to core/logger

### PR DESCRIPTION
**Pull request by pinebit**
Closes https://app.shortcut.com/chainlinklabs/story/36331/add-tests-core-logger

### CODE REVIEW

File: core/logger/fields_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+	w := f.With("bool", true, "float", 3.14)
+	assert.Len(t, w, 4)
+
+	t.Run("single", func(t *testing.T) {
+		t.Parallel()
+
+		assert.Panics(t, func() {
+			//lint:ignore SA5012 we expect panic here
+			_ = f.With("xyz")
```
[the remark this answers is not in the document]
Comment (pinebit) [reply core/logger/fields_test.go:79]: If this does not panic then `assert.Panics` would fail, no?

**Review by jmank88 — APPROVED** [review]

File: core/logger/passthrough_test.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+	ml.On("Fatal").Once()
+
+	ml.On("Tracef", mock.Anything).Once()
+	ml.On("Debugf", mock.Anything).Once()
+	ml.On("Infof", mock.Anything).Once()
+	ml.On("Warnf", mock.Anything).Once()
+	ml.On("Errorf", mock.Anything).Once()
+	ml.On("Criticalf", mock.Anything).Once()
+	ml.On("Panicf", mock.Anything).Once()
+	ml.On("Fatalf", mock.Anything).Once()
```
[the remark this answers is not in the document]
Comment (pinebit) [reply core/logger/passthrough_test.go]: Done

File: core/logger/prometheus_test.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	var m io_prometheus_client.Metric
+	err := c.Write(&m)
+	assert.NoError(t, err)
+	assert.Equal(t, v, int(m.GetCounter().GetValue()))
+}
+
+func repeat(f func(args ...interface{}), c int) {
+	for c > 0 {
+		f()
+		c--
```
[the remark this answers is not in the document]
Comment (pinebit) [reply core/logger/prometheus_test.go]: Done

File: core/logger/fields_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmank88) [reply core/logger/fields_test.go:79]: Yeah, but could the error could have some useful context for debugging the failure? nbd either way

**Review by jmank88 — APPROVED** [review]

**Pull request merged**